### PR TITLE
27 python311 in doc

### LINKED ISSUE #27: Specify in the documentation explicitly, that Python 3.11 has to be used
Maybe it is a good idea to state in the documentation that developers have to use Python 3.11 (not 3.10 and not 3.12) due to coldfront compatibility issues. Though present in the .toml file, one could make that explicit in the README.

### PR DESCRIPTION
**Pull request by HarmonicReflux**
# Description

Explicitly state `coldfront`'s `Python 3.11` dependency in `README.md` at the time of writing. 

Fixes #27 

## Type of change

- [ ] Documentation (non-breaking change that adds or improves the documentation)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (non-breaking, back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (whatever its nature)

## Key checklist

- [ ] All tests pass (eg. `python -m pytest`)
- [ ] The documentation builds and looks OK (eg. `python -m sphinx -b html docs docs/build`)
- [ ] Pre-commit hooks run successfully (eg. `pre-commit run --all-files`)

## Further checks

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added or an issue has been opened to tackle that in the future. (Indicate issue here: # (issue))

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,20 +42,24 @@ Versions can be restricted from updating within the `pyproject.toml` using stand
 
 To get started:
 
-1. Create and activate a [virtual environment](https://docs.python.org/3/library/venv.html):
+1. Create and activate a [virtual environment](https://docs.python.org/3/library/venv.html).
+   
+**Note:** It is important that this environment is created with `Python 3.11`. It cannot be any higher or lower Python version due to `coldfront` limitations at the time of writing. You may note we ensure this requirenment in the `pyproject.toml` file by `requires-python = "~=3.11"`, but wish to mention it again explicitly here. A tool like `pyenv` that has any Python version activated tht is different from `Python 3.11`, can bypass this requirenment and lead to side effects.
+
+Taking care of the note, we can then create a virual environment from the comman line like so:
```
Comment (AdrianDAlessandro) [inline README.md:49]:
```suggestion
Taking care of the note, we can then create a virtual environment from the command line like so:
```

File: README.md [hunk 2]
```diff
@@ -42,20 +42,24 @@ Versions can be restricted from updating within the `pyproject.toml` using stand
 
 To get started:
 
-1. Create and activate a [virtual environment](https://docs.python.org/3/library/venv.html):
+1. Create and activate a [virtual environment](https://docs.python.org/3/library/venv.html).
+   
+**Note:** It is important that this environment is created with `Python 3.11`. It cannot be any higher or lower Python version due to `coldfront` limitations at the time of writing. You may note we ensure this requirenment in the `pyproject.toml` file by `requires-python = "~=3.11"`, but wish to mention it again explicitly here. A tool like `pyenv` that has any Python version activated tht is different from `Python 3.11`, can bypass this requirenment and lead to side effects.
```
Comment (AdrianDAlessandro) [inline README.md:47]: Not entirely sure what the last sentence is meant to mean, but it is probably not necessary in any case
```suggestion
**Note:** It is important that this environment is created with `Python 3.11`. It cannot be any higher or lower Python version due to `coldfront` limitations at the time of writing. You may note we ensure this requirement in the `pyproject.toml` file by `requires-python = "~=3.11"`, but wish to mention it again explicitly here.
```

**Review by AdrianDAlessandro — APPROVED** [review]
Good, a couple of typos and suggested removing some unnecessary detail

**Review by SaranjeetKaur — APPROVED** [review]
Nice one!

**Pull request merged**